python/src/aiconfig/editor/client/src/components/AIConfigEditor.tsx

```diff
@@ -641,17 +637,9 @@ export default function EditorContainer({
                 type: "CONSOLIDATE_AICONFIG",
                 action: {
                   ...action,
-                  // Ensure we keep the prompt in a running state since this is an in-progress update
-                  isRunning: true,
```

Comment:
Maybe also remove this property from the CONSOLIDATE_AICONFIG action in aiConfigReducer

Reply:
Wasn't this added specifically to support streaming? It wasn't needed for the single awaited request run calls. Also, just noting that we still probably want to support non-streaming execution since external deployments might not have streaming implemented

Reply:
https://github.com/lastmile-ai/aiconfig/commit/010f499a7cea9b383292b0ff67315079510338de

> Fixed some bugs on the frontend along the way -- specifically, because we return the full aiconfig on every chunk update, we were resetting isRunning to false after the first chunk was received (by dispatching CONSOLIDATE_AICONFIG). Ideally, we update just the output chunk, but to fix this I've added a separate aiconfig_complete event to distinguish the finalized aiconfig from the in-progress one

I guess my main concern is will this be needed for gradio streaming?

Reply:
Ah that's a good point. Gradio would need the full updated value

Reply:
> Also, just noting that we still probably want to support non-streaming execution since external deployments might not have streaming implemented

We still have another CONSOLIDATE_AICONFIG call at the end after the stream outputs. 

> Wasn't this added specifically to support streaming? It wasn't needed for the single awaited request run calls.

I think you're right about this, if we return the entire AIConfig as a streamed response this will incorrectly make the `isRunning` param to false. However if we keep it here, we'll never terminate the `isRunning` state to false, because everything we return from the server is in stream format. 


https://github.com/lastmile-ai/aiconfig/assets/151060367/a6db0ff2-0390-4b2c-80e9-4c1364d79ae3



What I will do in a followup PR is add another streaming event called `aiconfig_stream` and have the `isRunning` there set to true (kind of the equivalent of `aiconfig` stream event today, while `aiconfig_complete` is now the same functionality as `aiconfig`). I just feel the names for that are a bit simpler and it's apparent that one is for streaming, while the other is for "regular" output as a "non-stream" (even if it's being implemented by our particular backend as a streaming event, other people can implement it as a non-stream event and know what it is supposed to do)